src/RobotSimulator.elm

```diff
@@ -0,0 +1,305 @@
+    in
+    position.x < screenWidth && position.x > 0 && position.y < screenHeight && position.y > 0
+
+
+moveByVelocity : GameObject -> GameObject
+moveByVelocity gameObject =
+    { gameObject | position = addVectors gameObject.position gameObject.velocity }
+
+
+moveShots : List GameObject -> List GameObject
+moveShots shots =
+    List.map moveByVelocity <| List.filter inBorders <| shots
+
+
+directionToVelocity : Direction -> Vector -> Vector
+directionToVelocity dir velocity =
+    case dir of
+        Right ->
+            Vector velocity.x 0
+
+        Left ->
+            Vector -velocity.x 0
+
+        Up ->
+            Vector 0 -velocity.y
+
+        Down ->
+            Vector 0 velocity.y
+
+
+moveRobot : Direction -> GameObject -> GameObject
+moveRobot dir robot =
+    let
+        position =
+            robot.position
+    in
+    asPositionIn robot <|
+        addVectors position <|
+            directionToVelocity dir robot.velocity
+
+
+formatModelOutput : Model -> ( Model, Cmd Msg )
+formatModelOutput model =
+    ( model, Cmd.none )
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    formatModelOutput <|
+        case msg of
+            Move dir ->
+                asRobotIn model <|
+                    let
+                        movedRobot =
+                            moveRobot dir <|
+                                asDirectionIn model.robot dir
+                    in
+                    if inBorders movedRobot then
+                        movedRobot
+
+                    else
+                        model.robot
+
+            Shoot ->
+                let
+                    direction =
+                        model.robot.direction
+
+                    robotPosition =
+                        model.robot.position
+                in
+                asShotsIn model <|
+                    GameObject (Vector (robotPosition.x + 7) (robotPosition.y + 7))
+                        (directionToVelocity direction shotVelocity)
+                        direction
+                        :: model.shots
+
+            Interval ->
+                asShotsIn model <|
+                    moveShots model.shots
+
+
+
+-- Subscriptions
+
+
+toDirection : String -> Msg
+toDirection keyCode =
+    case keyCode of
+        "ArrowRight" ->
+            Move Right
+
+        "ArrowLeft" ->
+            Move Left
+
+        "ArrowUp" ->
+            Move Up
+
+        "ArrowDown" ->
+            Move Down
+
+        " " ->
+            Shoot
+
+        _ ->
+            Move Right
+
+
+keyDecoder : Decode.Decoder Msg
+keyDecoder =
+    Decode.map toDirection (Decode.field "key" Decode.string)
+
+
+frameRate amount =
+    1 / amount
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Sub.batch
+        [ Time.every (frameRate 30) (always Interval)
+        , Browser.Events.onKeyDown keyDecoder
+        ]
+
+
+
+-- View
+
+
+toDegrees : Direction -> String
+toDegrees dir =
```

Comment:
If a symbol is used just once, there's no need to make it global. Just put it in a `let` expression where needed.